Lint additional four files in the tests folder

MVrachev opened the pull request:
Related to #1582 

**Description of the changes being introduced by the pull request**:

When preparing the pr which was going to resolve #1582 I had to rename the test files
related to the old codebase. That way I was able to exclude the old test files when linting
the `tests/` folder.
While I did that I realized I have to decide whether we want to lint and consequently rename
`utils.py`, `test_utils.py`, `simple_server.py`, and `aggregate_tests.py` after we remove the files testing the old code.
I decided that it's worth linting them for the following reasons:
- `utils.py` provides utilities that are used by our new code such as the `TestServerProcess` class
which handles server handling, logging, and the `run_sub_tests_with_dataset` decorator which
we used in multiple files.
- `test_utils.py` will be needed even after we remove the old test files as we want to have tests
that verify that `TestServerProcess` from `utils.py` is working as expected
- `simple_server.py` is the only helper for server startup used in the new code base and
more precisely to start the server in `test_updater_ng.py`
- `aggregate_tests.py` will be needed to aggregate the test files

That's why I decided to run the linters upon those files.
I tried to summarize the manual changes I had to make in the commit messages
as they weren't many.
Most of the changes are automatic and are result from `black` and `isort`.

**Please verify and check that the pull request fulfills the following
requirements**:

- [ ] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

Commit:
Apply linters on tests/utils.py

Apply all 4 linters on tests/utils.py, so we can lint it in future and
not rename and exclude it.
It's expected that most if not all code in tests/utils.py will be
useful even when we remove the tests file on the old code.
Keep in mind when reviewing that type annotations where already added
in https://github.com/theupdateframework/python-tuf/pull/1687/commits/e2deff3148514ceee7d9b3b1bc4ea3886e8b72b3

Black and isort changes where automatically made.
The only manual changes are:
- pylint disables
- function docstrings
- initializations of attributes of "TestServerProcess" __init__()
- additional asserts of the attributes types

Signed-off-by: Martin Vrachev <mvrachev@vmware.com>

Commit:
Apply linters on tests/test_utils.py

Apply all 4 linters on tests/test_utils.py, so we can lint it in the
future and not rename and exclude it.
As we are going to use part or most of tests/utils.py after we remove
the test files testing the old code, then we would need to keep the
test file testing utils.py - test_utils.py

Black and isort changes where automatically made.
The only manual changes are:
- pylint disable once in can_connect
- remove TestServerProcess.tearDown and instead clean it manually in the
tests, so we don't have to define server_process_handler as a class
variable and assert it's type in every test
- move can_connect outside TestServerProcess as it doesn't use self
anymore
- add type annotations
- function docstrings

Signed-off-by: Martin Vrachev <mvrachev@vmware.com>

Commit:
Apply linters on tests/simple_server.py

Apply all 4 linters on tests/simple_server.py, so we can lint it in the
future and not rename and exclude it.
As we are going to use part or most of tests/utils.py after we remove
the test files testing the old code, then we would need to keep the
simple_server.py file. It's currently used in tests/updater_ng.py
testing the new updater implementation.

Black and isort changes where automatically made.
The only manual changes are:
- pylint disable once in can_connect
- add type annotations
- simplifications around setting up the handler variable
- function docstrings

Signed-off-by: Martin Vrachev <mvrachev@vmware.com>

lukpueh commented on tests/simple_server.py:
```diff
@@ -25,40 +25,42 @@
+    Union[SimpleHTTPRequestHandler, QuietHTTPRequestHandler]
+] = QuietHTTPRequestHandler
 
-if use_quiet_http_request_handler:
-  handler = QuietHTTPRequestHandler
-else:
-  handler = SimpleHTTPRequestHandler
+if len(sys.argv) > 2:
+    handler = SimpleHTTPRequestHandler
```
For the purpose of this script the type annotation really seems like an overkill that impairs readability. Would it be problematic if we just did:
```python
if len(sys.argv) > 2 and sys.argv[2]:
  handler = QuietHTTPRequestHandler
else:
  handler = SimpleHTTPRequestHandler
```

MVrachev replied:
That way I receive the following two errors by `mypy`:
```
tests/simple_server.py:61: error: Cannot assign multiple types to name "handler"  without an explicit "Type[...]" annotation  [misc]

tests/simple_server.py:61: error: Incompatible types in assignment (expression has type "Type[SimpleHTTPRequestHandler]", variable has type "Type[QuietHTTPRequestHandler]")  [assignment]
```

I wanted to simplify it somehow but didn't come up with an idea of how to do it without a `mypy` error...
That's how I end up with this ugly line...

lukpueh replied:
I guess you could "declare" `handler` first. Looks pretty unpythonic to me, but a bit nicer then the long line above. 
```python
handler: Type[Union[SimpleHTTPRequestHandler, QuietHTTPRequestHandler]]

if len(sys.argv) > 2 and sys.argv[2]:
  handler = QuietHTTPRequestHandler
else:
  handler = SimpleHTTPRequestHandler
```

But I'm fine with leaving it as is.

MVrachev replied:
You are right that's better even if it's not the most pythonic way it's still more understandable.
Updated the code with that suggestion.

lukpueh commented on tests/utils.py:
```diff
@@ -86,253 +91,265 @@ def ignore_deprecation_warnings(module: str) -> Iterator[None]:
+                    + "!"
+                )
+
+            if line.startswith(expected_msg):
```
Out of curiosity, did `black` make this `elif` an `if`? Or did pylint flag it?

MVrachev replied:
It was `pylint` who flag it with:
`R1720: Unnecessary "elif" after "raise" (no-else-raise)`

lukpueh replied:
👍  Oh right, I remember that one.

lukpueh reviewed (approved):
Thanks, Martin! LGTM.

Commit:
Apply black on tests/aggregate_tests.py

All changes are automatically generated by black.
The other linters didn't find any errors in the file.
It's expected that most if not all code in aggregate_tests.py will be
useful even when we remove the tests file on the old code.

Signed-off-by: Martin Vrachev <mvrachev@vmware.com>

MVrachev commented:
I added a new commit linting `aggregate_tests.py` as we expect to use it even after we drop the old test files.
